File: imhere-api-gateway/src/main/java/com/kdongsu5509/imhereapigateway/filter/logger/AccessLogFormatter.java
package com.kdongsu5509.imhereapigateway.filter.logger;

import com.kdongsu5509.imhereapigateway.domain.AccessLog;
import org.springframework.stereotype.Component;

@Component
public class AccessLogFormatter {
    private final RequestLogBuilder requestBuilder;
    private final ResponseLogBuilder responseBuilder;

    public AccessLogFormatter(RequestLogBuilder requestBuilder, ResponseLogBuilder responseBuilder) {
        this.requestBuilder = requestBuilder;
        this.responseBuilder = responseBuilder;
    }

    public String format(AccessLog accessLog) {
        StringBuilder sb = new StringBuilder("\n");
        requestBuilder.build(accessLog, sb);
        responseBuilder.build(accessLog, sb);
        return sb.toString();
    }
}

File: src/main/kotlin/com/kdongsu5509/support/logger/AccessLogFormatter.kt
package com.kdongsu5509.support.logger

import org.springframework.stereotype.Component

@Component
class AccessLogFormatter(
    private val requestBuilder: RequestLogBuilder,
    private val responseBuilder: ResponseLogBuilder
) {
    fun format(accessLog: AccessLog): String {
        val sb = StringBuilder("\n")
        requestBuilder.build(accessLog, sb)
        responseBuilder.build(accessLog, sb)
        return sb.toString()
    }
}
